algorithms/src/signature/schnorr.rs

```diff
@@ -89,8 +89,7 @@ impl<G: ProjectiveCurve + CanonicalSerialize + CanonicalDeserialize> ToBytes for
     /// Writes the x-coordinate of the encryption public key.
     #[inline]
     fn write_le<W: Write>(&self, mut writer: W) -> IoResult<()> {
-        let affine = self.0.into_affine();
-        let x_coordinate = affine.to_x_coordinate();
+        let x_coordinate = self.0.to_x_coordinate();
```

Comment:
This is not always sound; it's sound only for the Edwards curves we're using. To make sure this works, I would change the trait bound to be `P: TEParameters`, and hard code the curve type to be Edwards